src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,318 @@
+     * @throws NoTaskIdException      if no taskID is provided.
+     * @throws InvalidTaskIdException If a non-numerical id is provided.
+     */
+    public void parseUnMark(String input) throws NoTaskIdException, InvalidTaskIdException {
+        String[] inputs = input.split(" ", 2);
+        if (inputs.length == 1) {
+            throw new NoTaskIdException();
+        } else {
+            String strIndex = inputs[1];
+            if (isNumber(strIndex)) {
+                int index = Integer.parseInt(strIndex) - 1; //because index starts from 1
+                if (tasks.isValidTaskId(index)) {
+                    tasks.unMarkTask(index);
+                }
+            } else {
+                throw new InvalidTaskIdException();
+            }
+        }
+    }
+
+    /**
+     * Handles the creation of a ToDo.
+     * Creates successfully if a description is provided.
+     *
+     * @param input user input.
+     * @throws NoDescException if no description is provided.
+     */
+    public void parseToDo(String input) throws NoDescException {
+        String[] inputs = input.split(" ", 2);
+        if (inputs.length == 1) {
+            throw new NoDescException();
+        } else {
+            if (inputs[1].isBlank()) {
+                throw new NoDescException();
+            } else {
+                //0 for unmarked, any other number for marked
+                ToDo toDo = new ToDo(0, inputs[1]);
+                tasks.addTask(toDo);
+            }
+        }
+    }
+
+    /**
+     * Handles the creation of a Deadline.
+     * Creates successfully if a description and an appropriate datetime is provided
+     * in the format 'deadline desc /by datetime' is provided.
+     *
+     * @param input user input.
+     * @throws NoDescException          if no description is provided.
+     * @throws InvalidDeadlineException if command is not of the correct format.
+     */
+    public void parseDeadline(String input) throws NoDescException, InvalidDeadlineException {
+        String[] inputs = input.split(" ", 2);
+        if (inputs.length == 1) {
+            throw new NoDescException();
+        } else {
+            String afterCommand = inputs[1];
+            String[] details = afterCommand.split(" /by ", 2);
+
+            if (details.length < 2) {
+                throw new InvalidDeadlineException();
+            } else {
+                String desc = details[0];
+                String date = details[1];
+
+                if (desc.isEmpty()) {
+                    throw new NoDescException();
+                } else {
+                    try {
+                        LocalDateTime dateTime = convertToDateTime(date);
+                        Deadline deadline = new Deadline(0, desc, dateTime);
+                        tasks.addTask(deadline);
+                    } catch (DateTimeParseException e) {
+                        ui.showInvalidDateFormat();
+                    }
+                }
+            }
+        }
```

Comment:
This part is quite heavily nested. Could you try to refactor the code to make it more flatter?